src/vcpkg/registries.cpp

```diff
@@ -1217,4 +1306,14 @@ namespace vcpkg
     {
         return try_parse_builtin_baseline(paths, "default");
     }
+
+    bool is_git_commit_sha(StringView sv)
+    {
+        static constexpr struct
+        {
+            bool operator()(char ch) { return ('0' <= ch && ch <= '9') || ('a' <= ch && ch <= 'f'); }
+        } is_lcase_ascii_hex;
```

Comment:
any reason not to use lambdas here?

```suggestion
        auto is_lcase_ascii_hex = [](char ch) { return ('0' <= ch && ch <= '9') || ('a' <= ch && ch <= 'f'); };
```

Reply:
No particular reason, but it's equivalent.